### PR TITLE
feat: Api to list available parent options

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

We need to choose only from dependent features that are in the same project and don't have any parents themselves.
This API returns a list of feature names that your feature can depend on. It will be used by the UI dropdown to populate the values.

### Important files

The interesting part is the DB query

### CODE REVIEW

File: src/lib/features/dependent-features/dependent-features-store.ts [hunk 1]
```diff
@@ -38,6 +38,23 @@ export class DependentFeaturesStore implements IDependentFeaturesStore {
         return rows.map((row) => row.child);
     }
 
+    async getParentOptions(child: string): Promise<string[]> {
```
Comment (kwasniew) [inline src/lib/features/dependent-features/dependent-features-store.ts:41]: algorithms:
* select your feature's project
* select all features from this project that are not children themselves

Comment (sjaanus) [reply src/lib/features/dependent-features/dependent-features-store.ts:41]: Is the naming deliberate? What are the options? getPossibleParentFeatures?

File: src/lib/features/dependent-features/dependent-features-store.ts [hunk 2]
```diff
@@ -38,6 +38,23 @@ export class DependentFeaturesStore implements IDependentFeaturesStore {
...
+            return [];
+        }
+        const rows = await this.db('features as f')
+            .leftJoin('dependent_features as df', 'f.name', 'df.child')
+            .where('f.project', result[0].project)
+            .andWhere('f.name', '!=', child)
+            .andWhere('df.child', null)
+            .select('f.name');
+
+        return rows.map((item) => item.name);
```
Comment (sjaanus) [inline src/lib/features/dependent-features/dependent-features-store.ts:55]: If the feature is already parent of child, should it still be returned?

File: src/lib/features/dependent-features/dependent-features-store.ts (hunk 1 above)
Comment (kwasniew) [reply src/lib/features/dependent-features/dependent-features-store.ts:41]: yup, I was thinking that parent options is a more elegant way of saying possible parent features :)

File: src/lib/features/dependent-features/dependent-features-store.ts (hunk 2 above)
Comment (kwasniew) [reply src/lib/features/dependent-features/dependent-features-store.ts:55]: if the parent is already a child then it shouldn't (` .andWhere('df.child', null)`). If the parent is already a parent it's a non-issue.

Comment (sjaanus) [reply src/lib/features/dependent-features/dependent-features-store.ts:55]: I am not fully sure when this query is being ran, but if child is feature A, if feature B is parent of A, should the query still return B?

Comment (kwasniew) [reply src/lib/features/dependent-features/dependent-features-store.ts:55]: if feature B is not a child of any other parent then yes. If your parent is a child of any other feature it should never be returned.

**Pull request merged**